src/librustc_resolve/diagnostics.rs

```diff
@@ -397,6 +397,76 @@ impl Bar {
 ```
 "##,
 
+E0417: r##"
+A static variable was referenced in a pattern. Example of erroneous code:
+
+```
+static FOO : i32 = 0;
+
+match 0 {
+    FOO => {} // error: static variables cannot be referenced in a
+              //        pattern, use a `const` instead
+    _ => {}
+}
+```
+
+Compiler needs to know the pattern value at compile's time, which is
+not possible with a `static` variable. So please verify you didn't
+misspell the variable's name or use a `const` instead. Example:
+
+```
+const FOO : i32 = 0;
+
+match 0 {
+    FOO => {} // ok!
+    _ => {}
+}
+```
+"##,
+
+E0424: r##"
+`self` keyword was used in a static method. Example of erroneous code:
+
+```
+struct Foo;
+
+impl Foo {
+    fn bar(self) {}
+
+    fn foo() {
+        self.bar(); // error: `self` is not available in a static method.
+    }
+}
+```
+
+Please verify you didn't forget to add `self` in your method's argument
```

Comment:
Please check if the method's argument list should have contained `self`, `&self`, or `&mut self`, and add it if so.